config/pmix.m4

```diff
@@ -890,6 +890,17 @@ AC_DEFUN([PMIX_SETUP_CORE],[
     CPP_INCLUDES="$(echo $cpp_includes | $SED 's/[[^ \]]* */'"$pmix_cc_iquote"'&/g')"
     CPPFLAGS="$CPP_INCLUDES -I$PMIX_top_srcdir/include $CPPFLAGS"
 
+
+    # We do not currently support the "lto" optimizer as it
+    # aggregates all the headers from our plugins, resulting
+    # in a configuration that generates warnings/errors when
+    # passed through their optimizer faze. We therefore check
+    # for the flag, and if found, output a message explaining
+    # the situation and aborting configure
+    _PMIX_CHECK_LTO_FLAG($CPPFLAGS, CPPFLAGS)
+    _PMIX_CHECK_LTO_FLAG($CFLAGS, CFLAGS)
```

Comment:
Might want to check $LDFLAGS and $LIBS, too.

I *think* the most appropriate location for `-flto` is in `$LDFLAGS` (it's a linker flag, after all), but people could choose to put it in any weird location (including $CFLAGS, $CPPFLAGS, $LIBS, etc.).

Reply:
It's actually "required" to be specified both at compile time and link time.
- If you pass it at link time only, it doesn't find any LTO btyecode to perform LTO on, and it's way too late to convert source code into bytecode.
- If you pass it at compile time only, the compiler *may* (probably does! I know gcc does, but I can't remember what clang does) choose to infer from the presence of bytecode in the object files, that it can do LTO. Object files can double-compile and produce both object code and bytecode, in which case you can choose to disable it at link time (-fno-lto) and it simply ignores the bytecode...

Basically the tl;dr here is that checking in CFLAGS is sufficient but checking in both won't hurt.